### ISSUE TITLE
Provide examples in documentation for common use cases

### ISSUE THREAD

**Issue by alexgleason** [matrix-org/matrix-js-sdk#430]
I see you have some example code in the repo, but it would be helpful to show some common tasks as well as how to do them with the SDK in the docs. As it stands it's difficult for a beginner to use this without first grasping all of the concepts of Matrix.

For instance:

* Sending a message
* Responding to a message
* Joining a room
* Adding images to a message

Basically some "getting started" tutorial sort of things would be great.

**Comment by ilyaigpetrov** [comment]
* Peeking into rooms as a guest
* Upgrading from guest to non-guest (with CAPTCHA maybe) and joining a room

**Comment by pixelass** [comment]
Yes. I literally have no idea how this works. I need to ask someone who has used it before and even then it's hard since nothing is documented.

**Comment by Ryuno-Ki** [comment]
@alexgleason @ilyaigpetrov @pixelass Where would you expect that documentation to live?
In code? On a website? In the README?

**Comment by alexgleason** [comment]
I was expecting to find examples here: http://matrix-org.github.io/matrix-js-sdk/0.12.1/index.html

But it's difficult to find this page as well (it's halfway down the README). Maybe this could be made more prominent and a section with examples could be added to it?

**Comment by richvdh** [comment]
Related: 
https://github.com/matrix-org/matrix-js-sdk/issues/430
https://github.com/matrix-org/matrix-js-sdk/issues/4449

**Comment by quochuydev** [comment]
Ref #4449 
I'd love to recommend my own repo if anyone is still looking.
https://github.com/quochuydev/matrix-e2e-emr
